W-11500706 d365 2.6 (#2031)
mulesoft/docs-connectors: microsoft-dynamics-365/2.6/modules/ROOT/pages/microsoft-dynamics-365-connector-reference.adoc

* DYNAMICS:OPERATION_FAILED
* DYNAMICS:PARSE_ERROR
* DYNAMICS:RETRY_EXHAUSTED
* DYNAMICS:TIMEOUT
* DYNAMICS:UNAUTHORIZED
* DYNAMICS:UNKNOWN

[[createMultiple]]
=== Create Multiple

`<dynamics:create-multiple>`

Creates multiple entities.

* Create New Entity - `+https://msdn.microsoft.com/en-us/library/gg328090.aspx+`
* Associate Entity Upon Creation - `+https://msdn.microsoft.com/en-us/library/mt607875.aspx#Anchor_4+`
* Batch Request - `+https://msdn.microsoft.com/en-us/library/mt607719.aspx+`

[%header,cols="20s,25a,30a,15a,10a"]
|===
| Name | Type | Description | Default Value | Required
| Configuration | String | Name of the configuration to use. | | x
| Logical Name a| String |  Logical name of the entity. It is always the schema name in lower case. |  | x
| Entities Attributes a| Array of Object |  List of the entities' attributes to create as a list of maps. |  `#[payload]` |
| Use Single Transaction a| Boolean |  Use a single transaction for creation. If the entity creation fails, the transaction is rolled back. |  `false` |
| Continue On Error a| Boolean |  Continue in case of error. The default value of `false` stops creating records after one create fails.

> You can use *Continue On Error* in conjunction with *Use Single Transaction*. Suppose you want to execute an operation with a payload of 10 records. This payload contains one broken record that will fail. This broken record is record number 5. The other nine records are certain to succeed. The following table shows the output depending on the values of *Continue On Error* and *Use Single Transaction*:

Reviewer: ```suggestion
You can use *Continue On Error* in conjunction with *Use Single Transaction*. Suppose you want to execute an operation with a payload of 10 records. This payload contains one broken record, number 5, that fails. The other 9 records succeed. The following table shows the output, which depends on the values of *Continue On Error* and *Use Single Transaction*:
```

Reviewer: Once you use a number instead of text for a number, you can use all numbers for consistency in a paragraph.

[%header%autowidth.spread]

> !===

Reviewer: For the Expected Result column, use a period for all entries for consistency.

! Use Single Transaction ! Continue on Error ! Expected result ! Successful records ! Failed records ! Unprocessed records
! False ! False ! No rollback. Stops processing records after a failed record. ! 1-4 ! 5 ! 6-10
! False ! True ! No rollback. Continues to process records even after a failed record. ! 1-4, 6-10 ! 5 ! N/A

> ! True ! False ! In case of failure, no records are commited ! N/A ! Failed records 1-10 ! N/A

Reviewer: ```suggestion
! True ! False ! If the connector fails to create an entity, no records are commited ! N/A ! Failed records 1-10 ! N/A
```

> ! True ! True ! In case of failure, no records are commited ! N/A ! Failed records 1-10 ! N/A

Reviewer: ```suggestion
! True ! True ! If the connector fails to create an entity,  no records are commited ! N/A ! Failed records 1-10 ! N/A
```

| Reconnection Strategy a| * <<reconnect>>
* <<reconnect-forever>> |  Retry strategy in case of connectivity errors. |  |
|===

==== Associated Configurations

* <<dynamics-config>>

==== Throws

* DYNAMICS:CONNECTIVITY
* DYNAMICS:INVALID_PAGE_SIZE
* DYNAMICS:NOT_FOUND
* DYNAMICS:OPERATION_FAILED
* DYNAMICS:PARSE_ERROR
* DYNAMICS:RETRY_EXHAUSTED
* DYNAMICS:TIMEOUT
* DYNAMICS:UNAUTHORIZED
* DYNAMICS:UNKNOWN

[[deleteMultiple]]
=== Delete Multiple

`<dynamics:delete-multiple>`

Deletes multiple entities.

* Delete an Entity - `+https://msdn.microsoft.com/en-us/library/mt607664.aspx+`
* Batch Request - `+https://msdn.microsoft.com/en-us/library/mt607719.aspx+`

[%header,cols="20s,25a,30a,15a,10a"]
|===
| Name | Type | Description | Default Value | Required
| Configuration | String | Name of the configuration to use. | | x
| Logical Name a| String |  Logical name of the entity. It is always the schema name in lower case. |  | x
| Ids a| Array of String |  List of IDs to delete. |  `#[payload]` |
| Use Single Transaction a| Boolean |  Use a single transaction for deletion. If the deletion fails, the transaction is rolled back. |  `true` |

> | Continue On Error a| Boolean |  Continue in case of error. The default value of `false` stops creating records after one create fails.

Reviewer 2: Continue in case of error. The default value is to stop deleting records after one delete failed.

> You can use *Continue On Error* in conjunction with *Use Single Transaction*. Suppose you want to execute an operation with a payload of 10 records. This payload contains one broken record that will fail. This broken record is record number 5. The other nine records are certain to succeed. The following table shows the output depending on the values of *Continue On Error* and *Use Single Transaction*:

Reviewer: Same edits as for the last Continue On Error description.

[%header%autowidth.spread]
!===
! Use Single Transaction ! Continue on Error ! Expected result ! Successful records ! Failed records ! Unprocessed records
! False ! False ! No rollback. Stops processing records after a failed record. ! 1-4 ! 5 ! 6-10
! False ! True ! No rollback. Continues to process records even after a failed record. ! 1-4, 6-10 ! 5 ! N/A

> ! True ! False ! In case of failure, no records are commited ! N/A ! Failed records 1-10 ! N/A

Reviewer: ```suggestion
! True ! False ! If the connector fails to delete an entity, no records are commited ! N/A ! Failed records 1-10 ! N/A
```

> ! True ! True ! In case of failure, no records are commited ! N/A ! Failed records 1-10 ! N/A

Reviewer: ```suggestion
! True ! True ! If the connector fails to delete an entity, no records are commited ! N/A ! Failed records 1-10 ! N/A
```

|===

==== Associated Configurations

* <<dynamics-config>>

==== Throws

* DYNAMICS:CONNECTIVITY
* DYNAMICS:INVALID_PAGE_SIZE
* DYNAMICS:NOT_FOUND
* DYNAMICS:OPERATION_FAILED
* DYNAMICS:PARSE_ERROR
* DYNAMICS:RETRY_EXHAUSTED
* DYNAMICS:TIMEOUT
* DYNAMICS:UNAUTHORIZED
* DYNAMICS:UNKNOWN

[[updateMultiple]]
=== Update Multiple

`<dynamics:update-multiple>`

Updates multiple entities.

* Update Entity - `+https://msdn.microsoft.com/en-us/library/mt607664.aspx+`
* Associate Entities Upon Update - `+https://msdn.microsoft.com/en-us/library/mt607875.aspx#Anchor_4+`
* Batch Request - `+https://msdn.microsoft.com/en-us/library/mt607719.aspx+`

[%header,cols="20s,25a,30a,15a,10a"]
|===
| Name | Type | Description | Default Value | Required
| Configuration | String | Name of the configuration to use. | | x
| Logical Name a| String |  Logical name of the entity. It is always the schema name in lower case. |  | x
| Entities Attributes a| Array of Object |  List of the entities' attributes to create as a list of maps. |  `#[payload]` |
| Use Single Transaction a| Boolean |  Use a single transaction for an update. If the update fails, the transaction is rolled back. |  `false` |

> | Continue On Error a| Boolean |  Continue in case of error. The default value of `false` stops creating records after one create fails.

Reviewer 2: Continue in case of error. The default value is to stop updating records after one update failed.

> You can use *Continue On Error* in conjunction with *Use Single Transaction*. Suppose you want to execute an operation with a payload of 10 records. This payload contains one broken record that will fail. This broken record is record number 5. The other nine records are certain to succeed. The following table shows the output depending on the values of *Continue On Error* and *Use Single Transaction*:

Reviewer: Same edits as before

[%header%autowidth.spread]
!===
! Use Single Transaction ! Continue on Error ! Expected result ! Successful records ! Failed records ! Unprocessed records
! False ! False ! No rollback. Stops processing records after a failed record. ! 1-4 ! 5 ! 6-10
! False ! True ! No rollback. Continues to process records even after a failed record. ! 1-4, 6-10 ! 5 ! N/A

> ! True ! False ! In case of failure, no records are commited ! N/A ! Failed records 1-10 ! N/A

Reviewer: ```suggestion
! True ! False ! If the connector fails to update an entity, no records are commited ! N/A ! Failed records 1-10 ! N/A
```
